### PR TITLE
docs: table sort

### PR DESCRIPTION
**Pull request by YardWill**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve performance.
- [ ] Type optimization

🐛 Bugfix

- [ ] Bug fix issue #0

🔧 Chore

- [ ] Test case
- [x] Docs / demos update
- [ ] CI / workflow
- [ ] Other

### 📝 Description

Add table sort demo

### 🖼️ Screenshot

|  Before  |  After  |
|----|----|
|  ❌  |  ✅  |

### CODE REVIEW

File: s2-site/examples/basic/table/index.en.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: 基础明细表格
```
Comment (lijinke666) [inline s2-site/examples/basic/table/index.en.md:2]: 英文

**Comment by xingwanying** [comment]
pr名不符合规范嗷～ Docs => docs

File: s2-site/examples/basic/table/demo/sort.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { SortMethod, TableSheet } from '@antv/s2';
+import '@antv/s2/dist/s2.min.css';
+
+fetch('../data/basic.json')
+  .then((res) => res.json())
+  .then((data) => {
+    const container = document.getElementById('container');
+    const canConvertToNumber = (sortKey: string) =>
```
Comment (lijinke666) [inline s2-site/examples/basic/table/demo/sort.ts:8]: sortKey: string 文档的编辑器 会把 ts 文件 转成 js，这个在 demo 编辑器 能跑过 ？

Comment (xingwanying) [reply s2-site/examples/basic/table/demo/sort.ts:8]: 嗳，demo没了？

**Review by xingwanying — APPROVED** [review]

**Pull request merged**